src/utils/iteratorUtils.ts

```diff
@@ -1,4 +1,4 @@
-export function forEach<K extends string, V>(
```

Comment:
Here I've removed the extends part, instead of switching from `string` to  `PrimaryKeyType`: i think these utilities should remain generic, let me know if there is a better option 😉

Reply:
These utilities operate on the model collections, so the keys will always equal `PrimaryKeyType` anyway. I'd say extending that type guards us against improper usage of these utilities, and it must be preserved.

Reply:
Updated